### PR TITLE
More clippy

### PR DESCRIPTION
**Pull request by kjeremy**
A few more clippy changes.

I'm a little unsure of the second commit. It's the trivially_copy_pass_by_ref lint and there are a number of places in the code we could use it if it makes sense.

### CODE REVIEW

File: crates/ra_assists/src/split_import.rs [hunk 1]
```diff
@@ -10,10 +10,7 @@ pub(crate) fn split_import(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assis
     let path = ast::Path::cast(colon_colon.parent())?;
     let top_path = successors(Some(path), |it| it.parent_path()).last()?;
 
-    let use_tree = top_path.syntax().ancestors().find_map(ast::UseTree::cast);
-    if use_tree.is_none() {
-        return None;
-    }
+    let _use_tree = top_path.syntax().ancestors().find_map(ast::UseTree::cast)?;
```
Comment (matklad) [inline crates/ra_assists/src/split_import.rs]: The original version seems a touch more readable to me

File: crates/ra_hir/src/code_model.rs [hunk 2]
```diff
@@ -299,7 +299,7 @@ impl Module {
             .collect()
     }
 
-    fn with_module_id(&self, module_id: CrateModuleId) -> Module {
+    fn with_module_id(self, module_id: CrateModuleId) -> Module {
```
Comment (matklad) [inline crates/ra_hir/src/code_model.rs]: :+1: we should have done this ages ago

File: crates/ra_assists/src/split_import.rs (hunk 1 above)
Comment (kjeremy) [reply crates/ra_assists/src/split_import.rs]: Fixed

**Comment by matklad** [comment]
Thanks!

bors r+

**Pull request merged**